src/transactions/typeTag/parser.ts

```diff
@@ -50,6 +54,7 @@ type TypeTagState = {
 
 export enum TypeTagParserErrorType {
   InvalidTypeTag = "unknown type",
+  UnexpectedGenericType = "unexpected generic type",
```

Comment:
do we need to document the `allowGenerics` option better?

Reply:
TBH I think it's fine with this, although maybe in the function doc comment we can note the option clearer.

The main thing for me that was confusing was the error didn't let me know it was a generic, so I didn't realize it had an understanding of generics. Once I realized that I just looked at the options args and it was clear how to enable them